### DISCUSSION TITLE
SK6812/WS2812B+ HyperHDR Issues

### DISCUSSION THREAD

**Discussion by BardoKudane** [awawa-dev/HyperHDR#498] [Q&A]
Recently bought supplies for a ambient TV-lighting setup:
Hardware:
BTF-LIGHTING RGBW RGB+Natural White SK6812 (Similar WS2812B) IP30
5V 15A 75W Power Supply
Raspberry Pi 3 Model B+ Board (3B+)
Rybozen 4K Audio Video Capture Card (Full HD 1080P)
Male to Female Jumper Wire
5.5mm x 2.1mm 24V Power Jack Female Barrel Jack Adapter
EDIT: Also bought BTF-LIGHTING WS2812B RGB 60 p/meter

Software:
Running HyperHDR v18.0.0.0

Things to take note.

LED's installed on a 65" Curved TV. Top 88/Bottom 88/ Left 50/ Right 50. (276 total)
LED's share a common ground/power with Micro USB. Grounds were butt-spliced/crimped together, with a single 14G copper wire going into negative. Same with the Power.
LED Data is connected to the GPIO 18 pin via jumper wire
Running HDMI from the ARC Out on my home theater receiver to HDMI input on the capture card. HDMI output from capture card to TV. Capture card is powered through the USB 3.0 from the Pi.
On initial setup, the first LED was flickering wildly between what appears to be white and green. Using the WS281x controller type, I changed the LED amount to the specified number listed above, I managed get the LED strip for the outer perimeter of the TV to illuminate; BUT still flickering white and green. At first I thought this might have been a grounding issue, however - when I installed the Hyperion mobile app on my phone and began toying with the color changer, I was able to get solid color to the back of the TV. Changing the color dynamically via the color wheel, will cause flickering again until I release finger from off my phone's screen.

With all that being said. I dont believe it is a grounding issue being transmitted to the LEDs if the lights are able to hold a solid color without flickering ( while using the Hyperion Mobile App). Power seems to be fine as all lights are extremely bright with no falloff at the end of the LEDs.

I am leaning a bit on there being a data issue, more specifically, using the WS281x controller type with SK6812 LEDs. But I have no knowledge on how to correct the issue if that IS the case.

Any guidance would be a great help. :thumbup:

**Comment by awawa-dev** [comment]
Hi
Direct driving Neopixel LED strip from Rpi is NOT supported here. Not to mention you're not using a 3.3V to 5V level shifter either, it's completely out of tech specs. It's a **bad** solution from the old Hyperion NG era, which I left only for backward compatibility reason and put a warning about it everywhere I could. Unfortunately, it's still ignored:

- Main page FAQ 
![obraz](https://user-images.githubusercontent.com/69086569/216813472-8f4de6b8-55d6-4b2e-a35a-e676dcd26610.png)
- Stick topic in the discussion [link](https://github.com/awawa-dev/HyperHDR/discussions/320)
![obraz](https://user-images.githubusercontent.com/69086569/216813498-7ddd64f1-a37e-4546-ad49-740c9300f3c5.png)

It *may* work for smaller LED strips when some other factors are 'luckily' satisfied. The longer LED strip (>200LEDs, usually it stops to work at all over 300LEDs), the higher density (60LEDs/meter or greater), RGBW LEDs strip type (more data to sent than RGB w812x) the harder for Rpi to emulate Neopixel signal and maintain proper timings, the greater risk it won't work well. Unfortunately do not have any hint for further diagnostic with such setup...most probably it's Rpi ws812x driver, maybe it's a missing level shifter, unstable power supply or too long data cable (EMI interference) or software configuration (anti-flickering filter in the smoothing but it's rather for the USB grabber as a source)
